pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -80,6 +81,11 @@ var diagMetrics types.MetricsMap
 var nimMetrics types.MetricsMap
 var zrouterMetrics types.MetricsMap
 
+// modified for real change detection, some fields zeroed out
+var modDPC types.DevicePortConfigList
+var modNodeStatus types.NodeAgentStatus
+var modDNS types.DeviceNetworkStatus
```

Comment:
These shouldn't be global variables. When comparing A and B you can clear the dont-care fields in both A and B and then do the compare.